cpp/oneapi/dal/detail/array_impl.hpp

```diff
@@ -88,18 +88,17 @@ class array_impl : public base {
         if (const auto& mut_ptr = std::get_if<shared>(&data_owned_)) {
             return mut_ptr->get();
         }
-        else {
-            const auto& immut_ptr = std::get<cshared>(data_owned_);
-            return immut_ptr.get();
+        else if (const auto& immut_ptr = std::get_if<cshared>(&data_owned_)) {
+            return immut_ptr->get();
         }
+        return nullptr;
```

Comment:
Can you please place a `ONEDAL_ASSERT` before this `return` statement? It should never be reachable.

Reply:
Done, thanks